config.go

```diff
@@ -185,6 +188,10 @@ type config struct {
 	MaxLogFiles    int    `long:"maxlogfiles" description:"Maximum logfiles to keep (0 for no rotation)"`
 	MaxLogFileSize int    `long:"maxlogfilesize" description:"Maximum logfile size in MB"`
 
+	LetsEncryptDir    string `long:"letsencryptdir" description:"The directory to store Let's Encrypt certificates within"`
```

Comment:
The config is starting to become quite big, what do you think about grouping auth related settings in their own namespace? :)

Reply:
I think it is a good idea, but not sure if it is considered a breaking change? @Roasbeef

Reply:
@halseth good point, this might be a good time to start taking advantage of the `lncfg` package. add a file like `lncfg/lets_encrypt.go` with `lncfg.LetsEncrypt` config struct. could at least start by having let's encrypt configs not directly on the main config?

Reply:
the namespacing works out quite nicely i think, `--letsencrypt.dir=xxx`, etc

Reply:
created the group `letsencrypt`